OBPIH-6388 github action for autogenerating release notes on new tag

EWaterman opened the pull request:
JIRA Ticket: https://pihemr.atlassian.net/browse/OBPIH-6388

Discovery Document: https://pihemr.atlassian.net/wiki/spaces/OB/pages/3056828417/DISCOVERY%3A+OBPIH-6388+Investigate+new+CI+solution

Adds a new GitHub Action that auto-generates the release notes whenever a new tag is pushed to the repo.

Notes:
- only runs if the tag starts with "vX.Y.Z" (Ex: 0.9.2, 0.9.2-hotfix...)
- The action can also be run manually in case it fails for any reason.
- If the job is re-run, it will replace the existing artifacts in the release

An example of running the Action manually:
![Screenshot from 2024-05-20 20-05-49](https://github.com/openboxes/openboxes/assets/6844376/b08e740d-fa94-4a49-adac-a24bc2a1ede8)

The release notes that are auto-generated once the action completes:
![Screenshot from 2024-05-20 19-16-12](https://github.com/openboxes/openboxes/assets/6844376/37aaa380-1b53-49d8-904c-a9fc92a560d9)

jmiranda commented:
@EWaterman As we discussed, you can merge this once you remove the testing part of this.

EWaterman commented:
I switched this to a new action so that it now displays the PRs correctly:

![Screenshot from 2024-05-31 09-42-31](https://github.com/openboxes/openboxes/assets/6844376/1e27a277-75ec-4a3c-962b-fd6939e8bff6)

I also removed the test step as mentioned and simplified the Gradle caching step.

A fun bonus of this change is that the release notes can separate our PRs now based on what they're tagged with. I've got an upcoming task to do some automatic PR tagging based on files changed and branch name, so hopefully that will be ready to demo soon!

I'm going to merge as is since this is a low risk change in that we can always delete the release notes that are generated or edit them.
